src/hdfmap/eval_functions.py

```diff
@@ -362,9 +404,42 @@ def format_hdf(hdf_file: h5py.File, expression: str, hdf_namespace: dict[str, st
     :param data_namespace: dict of {'variable name': value}
     :param replace_names: dict of {'variable_name': expression}
     :param default: returned if varname not in namespace
+    :param raise_errors: raise exceptions if True, otherwise return str error message as result and log the error
     :return: eval_hdf(f"expression")
     """
     expression = 'f"""' + expression + '"""'  # convert to fstr
```

Comment:
Does this work now that you are not `eval`-ing? Maybe something similar to `expression.format(**local())`?